### ISSUE TITLE
ResNet50 Permutation Spec Is Incorrect (Typo)

### ISSUE THREAD

**Issue by gstoica27** [samuela/git-re-basin#10]
Hi,

Thanks for your work and the accessibility of this repo! Very useful in our experiments :) 

I believe there is a typo on lines 145 & 146 in the weight_matching.py file: 
```python
shortcutblock = lambda name, p_in, p_out: {
      **conv(f"{name}/conv1", p_in, f"P_{name}_inner1"),
      **norm(f"{name}/norm1", f"P_{name}_inner1"),
      #
      **conv(f"{name}/conv2", f"P_{name}_inner1", f"P_{name}_inner2"),
      **norm(f"{name}/norm2", f"P_{name}_inner2"),
      #
      **conv(f"{name}/conv2", f"P_{name}_inner2", p_out), 
      **norm(f"{name}/norm2", p_out),
      #
      **conv(f"{name}/shortcut/layers_0", p_in, p_out),
      **norm(f"{name}/shortcut/layers_1", p_out),
  }
```

I believe you've accidentally referred to conv2 and norm2 twice. Instead the second mention should be conv3 and norm3:

```python
shortcutblock = lambda name, p_in, p_out: {
      **conv(f"{name}/conv1", p_in, f"P_{name}_inner1"),
      **norm(f"{name}/norm1", f"P_{name}_inner1"),
      #
      **conv(f"{name}/conv2", f"P_{name}_inner1", f"P_{name}_inner2"),
      **norm(f"{name}/norm2", f"P_{name}_inner2"),
      #
      **conv(f"{name}/conv3", f"P_{name}_inner2", p_out),
      **norm(f"{name}/norm3", p_out),
      #
      **conv(f"{name}/shortcut/layers_0", p_in, p_out),
      **norm(f"{name}/shortcut/layers_1", p_out),
  }
```

Thanks!

**Comment by samuela** [comment]
Hi @gstoica27, thanks so much for bringing this to our attention! Indeed, you're absolutely right that this is a typo! As it turns out (and thankfully for the sake of our camera ready!) this was actually pseudocode that didn't end up getting used in our final ResNet50 experiments. Due to having to switch machines between experiments, the ResNet50 results were implemented and run on a separate machine and with a different `PermutationSpec` implementation.

This was a very confusing, and I should've done a cleaner job with the code releaese. You're absolutely right for raising this issue. In an attempt to remedy this, we've merged #11 which should bring the source up-to-date and include the [correct ResNet50 `PermutationSpec` implementation](https://github.com/samuela/git-re-basin/blob/ef40098257ab97243930eba737d6dcb8edd5863e/src/weight_matching.py#L114-L171).

Apologies for the confusion!

**Comment by gstoica27** [comment]
Hi @samuela 

No worries at all! 

Thank you so much for adding the new permutation spec and code - it is so helpful for us :) 

Thanks so much again, we are also able to confirm that the implementation is working for our experiments on our end as well. :) 

Thanks again! 

All the best,
George
